File: js/tests/perfectNumber.test.js
const checkPerfect = require("../perfectNumber");

const data = [
  [6, true],
  [28, true],
  [496, true],
  [8128, true],
  [33550336, true],
  [12, false],
  [97, false],
  [481, false],
  [1001, false],
  [55555, false]
];

describe("checkPerfect", () => {
  describe("Return boolean if input is equal to sum of its factors", () => {
    test.each(data)("checkPerfect(%i) = %s", (num, result) => {
      let actual = checkPerfect(num);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/PerfectNumberTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class PerfectNumberTest
    {
        [Theory]
        [InlineData(6, true)]
        [InlineData(28, true)]
        [InlineData(496, true)]
        [InlineData(8128, true)]
        [InlineData(33550336, true)]
        [InlineData(12, false)]
        [InlineData(97, false)]
        [InlineData(481, false)]
        [InlineData(1001, false)]
        [InlineData(55555, false)]
        public void CheckPerfect_IntValueInput_ReturnBooleanIfSumOfFactorsIsEqualToInput(int num, bool expected)
        {
            var actual = PerfectNumber.CheckPerfect(num);

            Assert.Equal(expected, actual);
        }
    }
}